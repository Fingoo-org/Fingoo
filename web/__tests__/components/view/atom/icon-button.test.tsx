import { render } from '@testing-library/react';
import IconButton from '@/app/ui/components/view/atom/icon-button/icon-button';
import { DotsHorizontalIcon } from '@heroicons/react/solid';

describe('IconButton', () => {
  it('default 스타일의 아이콘 버튼을 렌더링한다.', () => {
    const { container } = render(<IconButton icon={DotsHorizontalIcon} color={'blue'} />);

    expect(container).toMatchSnapshot();
  });

  it('simple 스타일의 아이콘 버튼을 렌더링한다.', () => {
    const { container } = render(<IconButton icon={DotsHorizontalIcon} color={'blue'} variant={'simple'} />);

    expect(container).toMatchSnapshot();
  });

  it('outlined 스타일의 아이콘 버튼을 렌더링한다.', () => {
    const { container } = render(<IconButton icon={DotsHorizontalIcon} color={'blue'} variant={'outlined'} />);

    expect(container).toMatchSnapshot();
  });
});
