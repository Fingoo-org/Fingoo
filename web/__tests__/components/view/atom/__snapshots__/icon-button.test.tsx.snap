// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton default 옵션의 아이콘 버튼을 렌더링한다. 1`] = `
<div>
  <button
    class="px-1.5 py-1.5 text-blue-500"
  >
    <svg
      aria-hidden="true"
      class="h-5 w-5"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6 10a2 2 0 11-4 0 2 2 0 014 0zM12 10a2 2 0 11-4 0 2 2 0 014 0zM16 12a2 2 0 100-4 2 2 0 000 4z"
      />
    </svg>
  </button>
</div>
`;

exports[`IconButton outlined 스타일의 아이콘 버튼을 렌더링한다. 1`] = `
<div>
  <button
    class="px-1.5 py-1.5 bg-white bg-opacity-20 border-blue-500 ring-blue-300 ring-opacity-40 text-blue-500 hover:bg-blue-500 hover:bg-opacity-20 hover:text-blue-500 border ring-2 rounded-lg"
  >
    <svg
      aria-hidden="true"
      class="h-5 w-5"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6 10a2 2 0 11-4 0 2 2 0 014 0zM12 10a2 2 0 11-4 0 2 2 0 014 0zM16 12a2 2 0 100-4 2 2 0 000 4z"
      />
    </svg>
  </button>
</div>
`;
